server/re/src/icatGeneralMS.cpp

```diff
@@ -138,11 +138,14 @@ msiQuota( ruleExecInfo_t *rei ) {
  * \sa none
 **/
 int  msiSetResource( msParam_t* xrescName, ruleExecInfo_t *rei ) {
-    char *rescName;
+    char* rescName{static_cast<char *>(xrescName->inOutStruct)};
 
-    rescName = ( char * ) xrescName->inOutStruct;
+    if (rei->doi == nullptr) {
+        rodsLog(LOG_ERROR, "Cannot set resource, DOI not found.");
```

Comment:
Use the new logger.
```cpp
#include "irods/irods_logger.hpp"

namespace
{
    using log_msi = irods::experimental::log::microservice;
} // anonymous namespace

// Replace rodsLog with ...
log_msi::error("your message ...");
```

See https://docs.irods.org/4.3.1/developers/logging/ for more info on the logging library.

Reply:
What's the benefit of the anonymous namespace? Does it keep log_msi better localized?

Reply:
Correct. The compiler will mark the symbols as private to the translation unit. It's not required. It's just one way to do things.

If you prefer to move the alias into the function directly, that's perfectly fine too.

Reply:
I was mostly just curious, since I don't think I've seen it used for this before.  Done.